Add a state field to create utility pseudo-classes (#31643)
twbs/bootstrap: site/content/docs/5.0/utilities/api.md

  "float": null,
);
```

> ## Adding hover utilities

Reviewer: ```suggestion
## Hover utilities
```

> To add hover utility actions, you must add `hover: $enable-hover-utilites` to the desired utility:

Reviewer: ```suggestion
To add hover utilities, add `hover: $enable-hover-utilities` to the desired utility. This is globally set to `false` by default.
```

  "float": null,
);
```

## Adding pseudo-classes to utilities

> With the `state` option, pseudo-classes utility classes can be generated:

Reviewer 2: `pseudo-classes utility classes` sounds weird. Maybe use `utility pseudo-classes` or `pseudo-class utilities`?

Author: «pseudo-class utilities» sounds good. Thanks

```scss
$utilities: (
  "shadow": (
    property: box-shadow,
    state: hover focus,
    class: shadow,
    values: (
      null: $box-shadow,
      sm: $box-shadow-sm,
      lg: $box-shadow-lg,
      none: none,
    )
  )
);
```

Output:

```css

> .shadow-hover:hover {
>   box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15) !important;
> }

Reviewer: Per my above comment, I think just `.shadow:hover` makes the most sense. Here we'd be allowing folks to opt-in to any state(s) for any classes.

```suggestion
.shadow:hover {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15) !important;
}
```

Author: А как в таком случае задать hover? Классы shadow и псевдокласс shadow:hover никак не будут отличаться. Это не приведёт ни к какому эффекту. Например: блок с shadow-sm при hover нужно изменить на shadow-lg

```html
<div class="shadow-sm"></div>
```

```css
.shadow-sm:hover {
  /* ??? He will repeat his own styles */
}
```

Author: And it should look like this:

```html
<div class="shadow-sm shadow-lg-hover"></div>
```

or 

```html
<div class="shadow-sm shadow-hover-lg"></div>
```

Reviewer 2: > Per my above comment, I think just `.shadow:hover` makes the most sense. Here we'd be allowing folks to opt-in to any state(s) for any classes.

This confuses me a bit @mdo. This means adding the `shadow` class would only add a shadow on hover?

Reviewer: Hmm might have been a brain fart from me—for some reason I was thinking of these as adding to the existing utilities, but that's not right as y'all have mentioned. So, my apologies and carry on!

Author: I would suggest generating with this pattern

```
.#{$pseudo}-#{$property-class + $infix + $property-class-modifier}:#{$pseudo}
```

This will separate the utility and its states. The user will be able to work faster with pseudo-classes.

For instance:

```
.hover-shadow-sm
.hover-text-primary
.focus-border-info
```

etc

Reviewer: Problem with that is it breaks our conventions elsewhere. I think we'd have to do the state at the end. Thoughts @MartijnCuppens and @ffoodd?

Author: @mdo Then I propose to leave it as it was. Classes .shadow-sm-hover etc
